src/client.rs

```diff
@@ -16,6 +17,66 @@ pub fn handle_client_only_action(action: ActionClientOnly) -> Result<()> {
                 .spawn()?
                 .wait()?;
         }
+        ActionClientOnly::Edit { editor_arg, file } => {
+            // what editor to use
+            let editor_err = std::env::var("EDITOR");
+            let editor: String;
+            if editor_err.is_err() && editor_arg.is_none() {
+                eprintln!("EDITOR environment variable not set. Try running with -e '<editor>' or set the environment variable");
+                std::process::exit(1);
+            } else if editor_arg.is_some() {
+                editor = editor_arg.unwrap();
+            } else {
+                editor = editor_err?;
+            }
+            // what file to edit, the xml or the scss file
+            // This is so ugly because of this: https://github.com/rust-lang/rfcs/issues/372
+            let paths = config_path()?;
+            let xml_file: std::path::PathBuf = paths.0;
+            let scss_file: std::path::PathBuf = paths.1;
+            let path: std::path::PathBuf;
+            let file = file.unwrap_or_default();
+            if file == "xml" {
+                path = xml_file;
+            } else if file == "scss" {
+                path = scss_file;
+            } else {
+                eprint!("Edit the eww.xml file or the scss file? (X/s) ");
+                path = if input()?.to_lowercase() == "s\n" {
+                    scss_file
+                } else {
+                    xml_file
+                }
+            }
```

Comment:
```suggestion
            let (xml_file, scss_file) = config_path()?;
            let path = match file {
              Some("xml") => xml_file,
              Some("scss") => scss_file,
              None => {
                eprint!("Edit the eww.xml file or the scss file? (X/s) ");
                if input()?.to_lowercase() == "s\n" { scss_file }  else { xml_file }
              }
            }
```

Reply:
This code doesn't compile and i'm unsure on how to make it compile. mapping the option doesn't work because it uses data not owned by the current function.. i tried putting it in a cow (pun intended), but i can't do fn calls in match cases..